dds_web/api/schemas/user_schemas.py

```diff
@@ -151,11 +155,19 @@ def verify_new_email(self, value):
     def verify_and_get_invite(self, data, **kwargs):
         """Verifies that the email is in the invite table and in that case saves the invite info."""
 
+        form_email = data["email"]
+        token_email = data["token_email"]  # Originates from the token and not from the form
+
+        # Avoid a simple attacker scenarios where one is submitting a different email than the invite was sent to
+        if form_email != token_email:
+            flask.current_app.logger.warning(f"Email mismatch: {form_email} != {token_email}")
+            raise ddserr.InviteError(message="Form email and token email not the same")
+
         invite = models.Invite.query.filter(
             models.Invite.email == sqlalchemy.func.binary(data.get("email"))
         ).one_or_none()
         if not invite:
-            raise ddserr.InviteError
+            raise ddserr.InviteError(message="No invite found for this email at schema validation")
 
         data["invite"] = invite
 
```

Comment:
```
    def make_user(self, data, **kwargs):
        """Deserialize to an User object"""
        token = flask.session.get("invite_token")
        if not dds_web.security.auth.matching_email_with_invite(token, data.get("email")):
            raise ddserr.InviteError(message="Form email and token email are not the same")

        common_user_fields = {
            "username": data.get("username"),
            "password": data.get("password"),
            "name": data.get("name"),
        }

        # Create new user
        invite = data.get("invite")
        if invite.role == "Researcher":
            new_user = models.ResearchUser(**common_user_fields)
            # Currently no project associations
        elif invite.role in ["Unit Admin", "Unit Personnel"]:
            new_user = models.UnitUser(**common_user_fields)

            new_user.is_admin = invite.role == "Unit Admin"

            invite.unit.users.append(new_user)
        elif invite.role == "Super Admin":
            new_user = models.SuperAdmin(**common_user_fields)

        # Create new email and append to user relationship
        new_email = models.Email(email=data.get("email"), primary=True)
        new_user.emails.append(new_email)
        new_user.active = True

        db.session.add(new_user)

        # TODO Use the TKEK here via the token.
        # Idea is to handle it in auth.py together with project_user_keys.py
        flask.session.pop("invite_token", None)

        # Delete old invite
        db.session.delete(invite)

        # Save and return
        db.session.commit()

        return new_user
```

Reply:
Added token handling to the beginning, commented next to the tkek comment and brought token popping to here because we will be done with the invite token here.